### ISSUE TITLE
Crash when restarting browser after exiting via menu

### ISSUE THREAD

**Issue by srirambv** [brave/brave-browser#36826]
## Description 
Crash when restarting browser after exiting via menu

## Steps to reproduce 

   1. Launch browser
   2. Open menu and exit
   3. Relaunch browser, browser crashes
   4. Relaunch again no crash 
   5. Check `chrome://crashes`, no crash logs shown

## Actual result 

https://github.com/brave/brave-browser/assets/17010094/86c0e77a-c6c4-4aeb-9e10-98729d9f24c9

## Expected result
No crash when relaunching browser after exit via menu

## Issue reproduces how often 
Easy

## Version/Channel Information:

- Can you reproduce this issue with the current Play Store version? Yes when 1.64.x goes into release 
- Can you reproduce this issue with the current Play Store Beta version? Yes
- Can you reproduce this issue with the current Play Store Nightly version? Yes

## Device details
   - Install type (ARM, x86): ARM
   - Device type (Phone, Tablet, Phablet): Tablet
   - Android version: 13
   
## Brave version
1.64.105

### Website problems only
- Does the issue resolve itself when disabling Brave Shields? NA
- Does the issue resolve itself when disabling Brave Rewards? NA
- Is the issue reproducible on the latest version of Chrome? NA

### Additional information

cc: @deeppandya @tapanmodh @hffvld @Uni-verse @kjozwiak

**Comment by hffvld** [comment]
Hi @srirambv I can't reproduce the issue. Do I need to enable the flag as well?

------------------

https://github.com/brave/brave-browser/assets/128532965/2a88651e-57ee-432f-a6ce-a65bd74cc099

**Comment by srirambv** [comment]
I don't think you need to enable any flags. I see the crash on all channels. Maybe it could be something with Android 13. I see your tablet is on Android 14

**Comment by AlexeyBarabash** [comment]
I guess I experienced similar [at this Slack thread ](https://bravesoftware.slack.com/archives/C03MMJ4CHEZ/p1689184861134589?thread_ts=1689104915.068129&cid=C03MMJ4CHEZ)

Here are some details I found back then.

When `Exit` app menu item is pressed, these calls happen:
```
ApplicationLifetime.terminate()
ChromeLifetimeController.onTerminate()
```

`ChromeLifetimeController.onTerminate` does two things in parallel:
1. Enumerates all activities, calls their `finish()` and waits till all activities will be destroyed
2. Launches the watchdog Runnable with 1 sec timeout to perform final terminate actions.

When either pt1 or pt2 is completed, `BrowserRestartActivity` is started, which does
```
// Kill the main Chrome process.
Process.killProcess(mainBrowserPid);
// Kill this process.
Process.killProcess(Process.myPid());
```

For both `debug` and `release` builds I didn't see the watchdog runnable (pt2) was whenever invoked.
For `release` build pt1 takes ~0.6 sec, and then the browser process gets terminated properly.

But for the `debug` build pt1 takes ~7 sec to receive confirmation about the last activity is finished and do terminate. And if I launch browser from the shortcut, _not sure what exactly happens with the process, I think the old existing is re-used_, then after 7 sec it receives the confirmation about the finished activity and does `killProcess`. It looks like self terminate of new-launched process, but this is not a crash formally, so no callstack or tombstone in logcat.

I think, there is no need to do anything with it, as it works good on `release` and details of failure for `debug` are clear.
